### PR TITLE
feat: support ./{language}/*.json

### PR DESCRIPTION
**Pull request by NilsJacobsen**
Hey @samuelstroschein,

I added support for nested files.

Example: 
de/dashboard.json
de/landing.json

To still have unique ids i used the file name as a prefix.

### CODE REVIEW

**Comment by NilsJacobsen** [comment]
In addition I added a second example folder to test the other file structure. Now we have two test files for the different types of structures. `example01.test.ts` and `example02.test.ts`

File: src/index.ts [hunk 1]
```diff
@@ -67,10 +80,24 @@ export async function readResources(
       "{language}",
       language
     );
-    const json = JSON.parse((await args.$fs.readFile(resourcePath, "utf-8")) as string)
-    // reading the json, and flattening it to avoid nested keys.
-    const flatJson = flatten(json) as Record<string, string>;
-    result.push(parseResource(flatJson, language));
+    if((await args.$fs.stat(`${resourcePath.replace("/*.json", "")}`)).isFile()) {
```
Comment (samuelstroschein) [inline src/index.ts:83]: We won't have access to `stat` anymore with the upcoming https://github.com/inlang/inlang/pull/559 :(

Any way we can achieve the same effect with only `readFile`, `mkdir`, `writeFile` and `readdir`? See https://github.com/inlang/inlang/pull/559/files#diff-e0a2f5d5dc03af58ffcd338bb5eeeaad78aec678b17e294c25f0e6cfa1079b02R12

**Comment by samuelstroschein** [comment]
PS I love how flexible inlang is. We don't have namespace support and yet a plugin can simply add namespace support. Amazing. cc @ivanhofer who should love this

File: src/index.ts (hunk 1 above)
Comment (NilsJacobsen) [reply src/index.ts]: I replaced it with a try if the read fails it switches over to the catch statement and does the other.

**Comment by ivanhofer** [comment]
Another question that is unrelated to this PR but could benefit the whole resources plugin API:

Should we export a function `readRessource` instead of `readRessources`?
Advantages:
 - it is easier to implement for plugin authors as they don't need to write a loop
 - faster reads/writes
 	- we don't need to load all 30+ ressources and their namespaces of a repository if in the UI only 2 are selected
 	- we could process them in parallel. For disk writes probably unrelated, but `readResources` could also communicate with a DB or Server
 - less memory usage: just load the things we currently need

Disadvantages:
 - breaking change
   - not really if `defineConfig` wraps existing `readRessources` calls into a `readResource` function

File: src/index.ts [hunk 2]
```diff
@@ -80,12 +80,12 @@ export async function readResources(
       "{language}",
       language
     );
-    if((await args.$fs.stat(`${resourcePath.replace("/*.json", "")}`)).isFile()) {
+    try {
       const json = JSON.parse((await args.$fs.readFile(resourcePath, "utf-8")) as string)
```
Comment (samuelstroschein) [inline src/index.ts:84]: Will also break :D I hopefully merge `@inlang-git/fs` today. If that happens, you can be sure that the API works after upding `@inlang/core` here

**Comment by samuelstroschein** [comment]
> Should we export a function readRessource instead of readRessources?
Advantages:

> it is easier to implement for plugin authors as they don't need to write a loop
> faster reads/writes
> we don't need to load all 30+ ressources and their namespaces of a repository if in the UI only 2 are selected
> less memory usage: just load the things we currently need  

@ivanhofer  There is no benefit to speed/memory usage. We can just call `readResources(["en"])` if we only want one resource. In other words `readResources` supports reading only one resource but `readSingleResource` doesn't support reading multiple resources :)

> it is easier to implement for plugin authors as they don't need to write a loop

I counter the argument. Plugin authors have full control over how resources are read with `readResources` which is a better DX than avoiding a for loop. 


> we could process them in parallel. For disk writes probably unrelated, but readResources could also communicate with a DB or Server

`readResources` can process in parallel already if the plugin author does so.

**Comment by ivanhofer** [comment]
> @ivanhofer There is no benefit to speed/memory usage. We can just call `readResources(["en"])` if we only want one resource. In other words `readResources` supports reading only one resource but `readSingleResource` doesn't support reading multiple resources :)

`readResources` currently reads all ressources:

image

If `readResource` would be present, then `readResources` could be just a loop over `readResource` with all defined `languages`. If a plugin author want's to improve it, he could also export a dedicated `readResources` function.

**Comment by samuelstroschein** [comment]
> readResources currently reads all ressources:

My bad. You are right! An initial version of inlang used `readResources(languages)`. Then I decided to just pass the entire config because `inlang-plugin-json` required the knowledge of config properties such as `referenceLanguage`. 

The `writeResources` function still follows the pattern/is easily called to only write what changed. 

I expect that we need to introduce a filter option for `readResources` eventually or your proposed API change. Doesn't seem to be a critical performance issue rn though. THus, I propose to wait

**Comment by samuelstroschein** [comment]
@NilsJacobsen I had to reverse the merge. 

Your implementation was not able to pass a roundtrip test after upgrading to @inlang/core v0.7.

**Comment by NilsJacobsen** [comment]
@samuelstroschein Alright, I'm gonna open a new PR that works with the latest core version.

**Pull request merged**